### DISCUSSION TITLE
GNS3 Web UI feedback

### DISCUSSION THREAD

**Discussion by eantowne** [GNS3/gns3-web-ui#1350] [Ideas]
Caveat, some of these are things that should qualify as issues, but wanted to provide the feedback first in case the devs have some reasoning behind, and reduce the number of new issues.

* Enable GNS3 Logo button (top left corner) menu even without opening a project. This will also allow the removal of the “server” entry on the top bar

* What is the use case for GNS3VM in the Web-UI?
    * Remove from “Preferences” and add to “Settings” if this is actually needed

* “Logo Button->Settings->Updates->Check for updates” only opens a new browser tab/window to https://gns3.com/software - Should this be instructions on how to update the software (Server/VM)?

* What is the actual use of the Settings->Console Settings option?
    * Webclient-Pack manages client side desktop apps
    * What command can these variable be passed to?

* “Preferences” is only available from Server->specific server selection add to Logo button
    * Consolidation across servers for templates, or add server selection below “Preferences”
    * Hamburger (3 dot) menu for templates under a category does not have “Edit” option, Only “Delete” and “Copy”. Yes, you can click the name of the entry, but that is not visibly communicated to the user without moving the mouse over the name. The 3 dot menu, would be the default interaction method.
    * “Preferences” should be renamed to “Template Settings” or “Template Preferences” and GNS3 VM removed/moved

* Widgets/Project Map Settings icons:
    * Swap the purpose of these icons. The “Toggle topology/servers summary” icon looks more like a menu than a toggle
    * Nothing about the “Project Map Settings” icon communicates a purpose that relates. I guess it may be trying to say there are a bunch of check blocks, but that is not very intuitive

* Options to resize pop-up windows (configuration for a node as an example)

* Why is there an inner scroll bar for node configuration (general settings)?

**Comment by eantowne** [comment]
Enable GNS3 Logo button (top left corner) menu even without opening a project. This will also allow the removal of the “server” entry on the top bar:

I know I initially requested the addition of "Servers" to the top bar, but I think this is a more elegant solution, and provides consistency across all pages.

**Comment by grossmj** [comment]
We are focusing on v3 version of the web-ui. I don't think we will have time to make the changes on v2.x

> Enable GNS3 Logo button (top left corner) menu even without opening a project. This will also allow the removal of the “server” entry on the top bar

Agreed, we could have the menu on other pages. However, we need to move the project specific entries (project settings & map settings) somewhere else in the case, any idea?

![Screenshot from 2022-07-18 00-37-15](https://user-images.githubusercontent.com/4752614/179427530-752d3193-acab-45ea-b8c5-f32663d72e2c.png)

> What is the use case for GNS3VM in the Web-UI?

This has been removed in v3

> “Logo Button->Settings->Updates->Check for updates” only opens a new browser tab/window to https://gns3.com/software - Should this be instructions on how to update the software (Server/VM)?

You are right, this is a bit "light"...

> What is the actual use of the Settings->Console Settings option?

This has been removed in v3. The idea was to have an Electron based version of the web-ui but we gave up of it.

> “Preferences” is only available from Server->specific server selection add to Logo button

Server here means controller and templates are stored on the controller . We will rename the servers page to controllers page in v3.

> Hamburger (3 dot) menu for templates under a category does not have “Edit” option, Only “Delete” and “Copy”. Yes, you can click the name of the entry, but that is not visibly communicated to the user without moving the mouse over the name. The 3 dot menu, would be the default interaction method.

Right, we will fix this.

> “Preferences” should be renamed to “Template Settings” or “Template Preferences” and GNS3 VM removed/moved

Will be done too.

> Widgets/Project Map Settings icons

I will check if we can use better icons.

> Options to resize pop-up windows (configuration for a node as an example)

> Why is there an inner scroll bar for node configuration (general settings)?

Thanks, we will fix these issues too 👍

**Comment by eantowne** [comment]
Understood with the focus of effort being on v3.

As for the project specific menu options, why not only have them enabled when there is a project open? You could either disable/grey out those portions, or just only have them displayed when a project is open.

**Comment by grossmj** [comment]
> As for the project specific menu options, why not only have them enabled when there is a project open? You could either disable/grey out those portions, or just only have them displayed when a project is open.

Actually, I thought about this after I posted ;) I think that's what we will do 👍

**Reply by grossmj** [reply]
I have opened an issue for it: https://github.com/GNS3/gns3-web-ui/issues/1360
